src/merkle/smt/full/concurrent/mod.rs

```diff
@@ -478,6 +478,15 @@ fn build_subtree(
     tree_depth: u8,
     bottom_depth: u8,
 ) -> (UnorderedMap<NodeIndex, InnerNode>, SubtreeLeaf) {
+    #[cfg(debug_assertions)]
+    {
+        let mut seen_cols = BTreeSet::new();
+        for leaf in &leaves {
+            if !seen_cols.insert(leaf.col) {
+                panic!("Duplicate column found in subtree: {}", leaf.col);
+            }
+        }
+    }
```

Comment:
This situation could happen only if we test/benchmark this method individually, right? That is, if we go through our public APIs, this assertion should never be violated - right? If so, I'd add a comment describing this.

Also, we could write the loop body as:
```rust
assert!(seen_cols.insert(leaf.col), "Duplicate column found in subtree: {}", leaf.col);
```